components/feature/download/README.md

```diff
@@ -0,0 +1,69 @@
+# [Android Components](../../../README.md) > Feature > Download
+
+Feature implementation for apps that want to use [Android downloads manager](https://developer.android.com/reference/android/app/DownloadManager).
+
+## Usage
+
+### Setting up the dependency
+
+Use gradle to download the library from JCenter:
+
+```Groovy
+implementation "org.mozilla.components:feature-download:{latest-version}"
+```
+### DownloadManager
+It is our representation to of the [Android downloads manager](https://developer.android.com/reference/android/app/DownloadManager). It allows you to download files and be subscribed for download events.
+
+```kotlin
+       val ctx = context.applicationContext
+        val manager = DownloadManager(ctx)
+
+        val download = Download(
+                "http://ipv4.download.thinkbroadband.com/5MB.zip",
+                null, "application/zip", 5242880,
+                "Mozilla/5.0 (Linux; Android 7.1.1) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Focus/8.0 Chrome/69.0.3497.100 Mobile Safari/537.36"
+        )
+
+        val onDownloadComplete: (Long) -> Unit = { downloadID ->
+            //Show some UI.
+        }
+
+        val refererURL = "https://www.mozilla.org/"
+        val cookie = "yummy_cookie=choco"
+
+        val downloadId = manager.download(download,
+                onDownloadComplete /*Optional*/,
+                refererURL /*Optional*/,
+                cookie /*Optional*/
+        )
+
+        /*Adding a listener to download in process
+        * If the download I've already had a listener
+        * it will be updated with the new one*/
+        manager.registerListener(downloadId, onDownloadComplete)
+
+        /*Unregistering all listeners*/
+        manager.unregisterListeners()
+```
```

Comment:
nit: I see the `DownloadManager` class more as an implementation detail. Do you think there's a need to use it in a standalone way? Anyhow, I'd at least move this to the end and explain `DownloadManagerFeature` first. :)

Reply:
Done!
I improved the documentation and put `DownloadsFeature` first.
I added the documentation of `DownloadManager` someone has a different use case that is not cover by `DownloadsFeature`.